Subject: change die_chain from atomic to raw notifiers
From: Ingo Molnar <mingo@elte.hu>

atomic notifier chains are using rcu_read_lock()/unlock(), but those
are not NMI-safe in -rt - so switch these chains to raw notifiers.

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 kernel/die_notifier.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

Index: linux-2.6.23.1-rt5/kernel/die_notifier.c
===================================================================
--- linux-2.6.23.1-rt5.orig/kernel/die_notifier.c
+++ linux-2.6.23.1-rt5/kernel/die_notifier.c
@@ -5,7 +5,7 @@
 #include <linux/kdebug.h>
 
 
-static ATOMIC_NOTIFIER_HEAD(die_chain);
+static RAW_NOTIFIER_HEAD(die_chain);
 
 int notify_die(enum die_val val, const char *str,
 	       struct pt_regs *regs, long err, int trap, int sig)
@@ -19,19 +19,19 @@ int notify_die(enum die_val val, const c
 
 	};
 
-	return atomic_notifier_call_chain(&die_chain, val, &args);
+	return raw_notifier_call_chain(&die_chain, val, &args);
 }
 
 int register_die_notifier(struct notifier_block *nb)
 {
 	vmalloc_sync_all();
-	return atomic_notifier_chain_register(&die_chain, nb);
+	return raw_notifier_chain_register(&die_chain, nb);
 }
 EXPORT_SYMBOL_GPL(register_die_notifier);
 
 int unregister_die_notifier(struct notifier_block *nb)
 {
-	return atomic_notifier_chain_unregister(&die_chain, nb);
+	return raw_notifier_chain_unregister(&die_chain, nb);
 }
 EXPORT_SYMBOL_GPL(unregister_die_notifier);
 
